### ISSUE TITLE
Test on VOC2012 error

### ISSUE THREAD

**Issue by moothes** [weiliu89/caffe#551]
Hello, I developed a new network base on SSD and want to test on [VOC2012](http://host.robots.ox.ac.uk:8080/). However, all my submissions got evaluation failed with the reason "**Results were not generated properly**". My detection results are generated by the same code as SSD. I have test it on VOC07 by myself and it was fine. The original results like:

![qq 20170423093411](https://cloud.githubusercontent.com/assets/17693547/25309857/5806375e-280a-11e7-919c-811759c9708c.png)

I checked the sample results from the evaluation server after I got the evaluation failed. I thought it may be wrong format or something, so I modified the results: 

![xu_ak i4svt6yr k o x](https://cloud.githubusercontent.com/assets/17693547/25309691/2992e8c6-2806-11e7-8484-307f12a25a06.png)

I replaced all '0' in the bounding box offsets with '1' and forced confidences and offsets to keep 6 decimal places as sample results. However, I got evaluation failed again. Visualization of the bounding boxes is correct too. I don't know what's wrong with my results. I miss anything?

**Comment by moothes** [comment]
I just checked about the VOC2012 test set and found that I have 16135 images for test, while only 10991 images in SSD paper. I downloaded the test set from the evaluation server. For verification, I downloaded VOC2012 test set mirror from [YOLO](https://pjreddie.com/projects/pascal-voc-dataset-mirror/) and it have 16135 images too. 

I futher checked about the training set. I got VOC07 train(5011 images), VOC07 test(4952 images) and VOC12 train(17125), and 27088 images in total. However, SSD paper indicated it contains 21503 images. The official documentation also said it contains 11540 images in VOC12 train set, coincident with the paper. I think I really downloaded these data from the server. 

Is this the reason why my evaluation failed? And where can I download the correct datasets?

**Comment by moothes** [comment]
The problem have been solved. 
I mistakenly believe that all images are training examples. The ImageSets folder contain images list for each task. After I picked out all images for detection, the total number is match up with the paper. And also get the test result on VOC2012 successfully.

**Comment by foralliance** [comment]
HI,
@moothes 
how do u test voc2012??
the code "score_ssd_pascal.py" Seems to fit only voc2007. because for voc2012, there is no VOC0712Plus_test_lmdb

**Comment by moothes** [comment]
Hi, @foralliance .
I used the SSD network to generate detection results on VOC2012 test set and submitted results to [http://host.robots.ox.ac.uk:8080/](url).
I do not have any anotations of VOC 2012 test.

**Comment by moothes** [comment]
Sorry, the url is [http://host.robots.ox.ac.uk:8080/](url)

**Comment by moothes** [comment]
Sorry, the url is http://host.robots.ox.ac.uk:8080

**Comment by foralliance** [comment]
HI, @moothes 
How do u test??
Using the test file in [refinedet,](https://github.com/sfzhang15/RefineDet) I also got the result. But how do you test if you only use the code provided by SSD? specifically, what is your code?
Many thanks

**Comment by DL-ljw** [comment]
Hi, @moothes 
sorry for bothering you. Can you tell me how I can test on VOC2012, the default is 2007.

**Comment by moothes** [comment]
@foralliance @DL-ljw . I tested it long time ago, so I forget the details. I make LMDB file and run my model on it. Then I get the result files and upload them to the evaluation server.
